server/controllers/userController.js

```diff
@@ -63,3 +67,57 @@ export const createUserController = async (req, res) => {
     );
   }
 };
+
+
+export const userSignIn = async (req, res) => {
+  const { usernameOrEmail, password  } = req.body;
```

Comment:
I would use username and email  fields in body, then create a logic that takes username or email.